<launch> 
<node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
  <param name="frequency" value="10"/>
  <param name="sensor_timeout" value="2.0"/>
  <param name="two_d_mode" value="true"/>  
  <param name="map_frame" value="map"/>
  <param name="odom_frame" value="odom"/>
  <param name="base_link_frame" value="base_link"/>
  <param name="world_frame" value="odom"/> 
  <param name="print_diagnostics" value="true"/>
  <param name="debug" value="false"/>
  <param name="debug_out_file" value="/tmp/debug.txt"/>

  <param name="publish_tf" value="true"/>
  <param name="imu0" value="/mavros/mavros/imu/data_raw"/>
  <rosparam param="imu0_config">[false, false, false,
                                true,  true,  true,
                                false,  false,  false,
                                true,  true,  true,
                                true,  true,  true]</rosparam>
  <param name="imu0_differential" value="false"/>   <!-- was false -->
  <param name="imu0_remove_gravitational_acceleration" value="true"/>

  <param name="odom0" value="/my_robot/odometry/gps"/>
  <!-- fuse X and Y position, yaw, XË™, and yawË™. -->
  <rosparam param="odom0_config">[true,  true,  true,
                                false, false, false,
                                false, false, false,
                                false, false, false,
                                false, false, false]</rosparam>

  <param name="odom0_differential" value="false"/>  <!-- was false -->

</node>

<node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" respawn="true" output="screen">
    <param name="magnetic_declination_radians" value="0"/>
    <param name="broadcast_cartesian_transform" value="true"/>
    <param name="wait_for_datum" value="true"/>
    <param name="publish_filtered_gps" value="true"/>
    <remap from="/gps/fix" to="/mavros/mavros/global_position/raw/fix"/>
    <remap from="/odometry/gps" to="/my_robot/odometry/gps"/>
    <rosparam param="datum">[40.8631, 29.2598, 0]</rosparam>

</node>
</launch>